RFCs/FS-1061-Allow-intrinsic-type-extensions-for-provided-types.md

```diff
@@ -0,0 +1,122 @@
+
+# Detailed design
+[design]: #detailed-design
+
+The compiler needs to be modified so that instead of an error being generated when a provided type is extended the type is extended:
+
+```
+Error FS0192: internal error: FindTypeDefBuilder: <Method> not found (FS0192) (<Application>)
+```
+
+Instead the type is extended in the same way as a normal type extension.
+
+The proposed systax is the same as the existing intrinsic type extension syntax:
+
+Example code short form:
+
+```fsharp
+type t = BasicProvider.Generative.MyType with
+    member x.Foo = 42
+```
+
+Example code long form:
+
+```fsharp
+type t = BasicProvider.Generative.MyType
+
+type t with 
+    member x.Foo = 42
+```
+
+Given that `MyType` is just a plain type with a default constructor the resulting generated code wold look like this:
+```
+type t =
+  class
+    new : unit -> t
+    member Foo : int
+```
+
+An addition change to languge specification also needs to be changed so that an override can be applied as an intrinsic type extension too.   Imagine that BasicProvider.Generative.MyType has an abstract method as follows:
+
+```fsharp
+abstract member Rotate: float -> unit
+```
+
+It would be natural to describe this as follows:
+
+```fsharp
+type t = BasicProvider.Generative.MyType
+
+type t with 
+    member x.Foo = 42
+    override x.Rotate _ = ()
+```
+
+This would result in the following warning:
+```
+[FS0060] Override implementations in augmentations are now deprecated. Override implementations should be given as part of the initial declaration of a type.
+```
+
+So the final part of this  proposal is to allow override implementations in provided types **only**, as it is not always possible to provide the implementation as part of the initial declaration of the type especially if you are only the consumer of the type provider. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The only drawback is the complexity of the addition, but the addition makes provided types a lot more flexible especially as F# has no partial types, the olny alternative without this is inheriting the provided types and adding implementations there which is not ideal.
+
+# Alternatives
+[alternatives]: #alternatives
+
+The alternative is to add features to the provided type by alternative means such as inheriting from the provided type or by augmenting another type which deferrs to the provided type internally.  
+
+Another alternative would be to add partial classes to F# which is really what intrinsic type extensions are albeit limited to extensions in the same file.
```

Comment:
Nit: Probably worth adding that this would be an expensive alternative, with consequences that are far-reaching (e.g., changing how Go to Definition works in editor tooling).

Reply:
done